src/openvic-simulation/map/Map.cpp

```diff
@@ -493,34 +493,48 @@ bool Map::load_map_images(fs::path const& province_path, fs::path const& terrain
 	std::vector<fixed_point_map_t<TerrainType const*>> terrain_type_pixels_list(provinces.size());
 	bool ret = true;
 	std::unordered_set<colour_t> unrecognised_province_colours;
+	
+	std::vector<fixed_point_t> pixels_per_province(provinces.size());
+	std::vector<fixed_point_t> x_sum_per_province(provinces.size());
+	std::vector<fixed_point_t> y_sum_per_province(provinces.size());
 
 	for (size_t y = 0; y < height; ++y) {
 		for (size_t x = 0; x < width; ++x) {
-			const size_t idx = x + y * width;
+			const size_t pixel_index = x + y * width;
+			const colour_t province_colour = colour_at(province_data, pixel_index);
+			
+			const Province::index_t province_index = get_index_from_colour(province_colour);
```

Comment:
`get_index_from_colour` requires searching through a big list of colours, which is an expensive operation. All of the code between here and `set_terrain:` is precisely to *avoid* needing to call `get_index_from_colour` unless absolutely necessary (by checking the provinces to the left and above in case they're the same colour, and so same index). Please move this back below `set_terrain` so it can use the efficiently calculated province index instead of forcing a costly lookup and making the `if (x > 0)` and `if (y > 0)` optimisations pointless.